admin/c2cgeoportal_admin/views/layers_wms.py

```diff
@@ -50,11 +53,24 @@ def index(self):
     def grid(self):
         return super().grid()
 
+    def _add_convert_url_to_view(self, view):
+        convert_url = self._request.route_url(
+            'layers_wmts_from_wms',
+            table='layers_wmts',
+            wms_layer_id=self._request.matchdict.get('id'),
+            action='from_wms')
+        view['actions'].append({'url': convert_url, 'label': _('Convert to wmts')})
+        return view
+
     @view_config(route_name='c2cgeoform_item',
                  request_method='GET',
                  renderer='../templates/edit.jinja2')
     def view(self):
-        return super().edit()
+        pk = self._request.matchdict.get('id')
+        if pk == 'new':
+            return super().edit()
+        else:
+            return self._add_convert_url_to_view(super().edit())
```

Comment:
I would prefer something like :
```
resp = super().edit()
if not self.is_new():
  resp['actions'].append(self.convert_url())
```
Note that you could add some helpers in abstract views : self.is_new()

Reply:
not sure we need a convert_url() method, we can directly integrate the route_url call here as it not used somewhere else.